### ISSUE TITLE
PythonPath Manager cannot browse *all* system directories on OS X

### ISSUE THREAD

**Issue by dsterl** [spyder-ide/spyder#3330]
## Description of your problem
Path manager does not show system directories that are hidden by default on OS X (like /usr/local) but are heavily used by developers.

**What steps will reproduce the problem?**

1. Open Python path manager 
2. Attempt to add a path like /usr/local/lib/Python2.7/site-packages
3. 

**What is the expected output? What do you see instead?**
Expect path manager to allow user to browse all directories on shown by Finder on system. 

**Please provide any additional information below**

## Versions and main components

* Spyder Version: 2.3.8
* Python Version: 2.7.12
* Operating system: 10.11.5

## Dependencies

Please go to the menu entry `Help > Optional Dependencies` (or
`Help > Dependencies`), press the button `Copy to clipboard`
and paste the contents below:
IPython >=1.0      :  4.0.1 (OK)
jedi >=0.8.1;<0.9.0:  None (NOK)
matplotlib >=1.0   :  1.5.0 (OK)
pandas >=0.13.1    :  0.17.1 (OK)
pep8 >=0.6         :  1.6.2 (OK)
psutil >=0.3       :  3.3.0 (OK)
pyflakes >=0.5.0   :  1.0.0 (OK)
pygments >=1.6     :  2.0.2 (OK)
pylint >=0.25      :  1.4.4 (OK)
qtconsole >=4.0    :  4.1.1 (OK)
rope >=0.9.2       :  0.10.2 (OK)
sphinx >=0.6.6     :  1.3.1 (OK)
sympy >=0.7.3      :  0.7.6.1 (OK)
zmq >=2.1.11       :  15.1.0 (OK)

**Comment by ccordoba12** [comment]
> Attempt to add a path like /usr/local/lib/Python2.7/site-packages

You shouldn't do this at all, unless you're using a Spyder version installed in `/usr/local`, like (for example) a Homebrew version. Mixing Python interpreters (i.e. the one that comes with Spyder and others in your system) is a sure cause for ugly crashes! :-)

The right way to use a different Python is for you to change your used interpreter. A list of steps to do it is mentioned here:

http://stackoverflow.com/questions/19795469/importing-arcpy-module-into-anacondas-spyder/19813292#19813292

----

If you still want to show hidden files in OS X in open dialogs, you need to follow these steps:

http://macs.about.com/od/usingyourmac/qt/opensavehide.htm

**Comment by ccordoba12** [comment]
I'm closing this issue because it's not a problem related to Spyder.

**Comment by dsterl** [comment]
Carlos,
Thanks for the quick reply! I am a python newbie setting up a scientific python stack from scratch. Spyder is fantastic, but I’ve been a bit frustrated trying to get my python path set properly so that I can use packages that I have installed on my system. I am using an home-brew python installation and custom packages developed by others in my group so all of my python packages are usr /usr/local rather than in the system library.

Thanks for the link below — I found it yesterday and I did successfully change my python interpreter, but then iPython kernel stopped working (it says “connecting to kernel” but never connects) so I changed it back to one shipped with Spyder and focused on changing the default python path instead.

I like the idea of switching python interpreter —how can I do this and still use Ipython in Spyder?

Best,

David
